src/api_server/src/lib.rs

```diff
@@ -146,6 +154,25 @@ impl ApiServer {
             .map_err(anyhow::Error::from)
     }
 
+    #[cfg(target_arch = "s390x")]
+    fn tls_config(&self) -> Result<SslAcceptorBuilder> {
+        use openssl::ssl::{SslAcceptor, SslMethod, SslFiletype};
+
+        let cert_file = self.certificate
+            .clone()
+            .ok_or_else(|| anyhow!("Missing certificate"))?;
+
+        let key_file = self.private_key
+            .clone()
+            .ok_or_else(|| anyhow!("Missing private key"))?;
+
+        let mut builder = SslAcceptor::mozilla_modern(SslMethod::tls()).map_err(|e| anyhow!(e.to_string()))?;
+        builder.set_private_key_file(key_file, SslFiletype::PEM).map_err(|e| anyhow!(e.to_string()))?;
+        builder.set_certificate_chain_file(cert_file).map_err(|e| anyhow!(e.to_string()))?;
```

Comment:
I'm not sure about this. `anyhow` gives a simple way to convert errors using `?`, could you try
```
let mut builder = SslAcceptor::mozilla_modern(SslMethod::tls())?;
builder.set_private_key_file(key_file, SslFiletype::PEM)?;
builder.set_certificate_chain_file(cert_file)?;
```